pkg/cmd/init.go

```diff
@@ -0,0 +1,361 @@
+/*
+Copyright 2022 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/spf13/cobra"
+	yamlv2 "gopkg.in/yaml.v2"
+
+	"k8c.io/kubeone/examples"
+	kubeonev1beta2 "k8c.io/kubeone/pkg/apis/kubeone/v1beta2"
+	"k8c.io/kubeone/pkg/fail"
+	"k8c.io/kubeone/pkg/yamled"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/yaml"
+)
+
+type initProvider struct {
+	terraformPath string
+	inTree        bool
+	cloudConfig   string
+	csiConfig     string
+}
+
+var (
+	defaultInitVersion = "x.y.z"
```

Comment:
I think this should be some real version so that we don't run into problems if someone uses `go install` (which should be rare, but we never know for sure).

Reply:
I'd then propose for such cases to keep it emptystring

Reply:
done

Reply:
> I'd just add some validation that would fail if it's empty.

why should it fail? This version being empty doesn't prevent manifest generation. Yes the version.kubernetes would be an empty line, about which user will be reminded to by the `kubeone apply` validation.